src/client/client.ts

```diff
@@ -583,8 +458,93 @@ export class DeskproClient implements IDeskproClient {
     return new EntityAssociation(this, name, entityId);
   }
 
-  public oauth2(): IOAuth2 {
-    return new OAuth2(this);
+  public async startOauth2Local(
+    authorizeUrlFn: (data: { state: string, callbackUrl: string, codeChallenge: string }) => string,
+    codeAcquisitionPattern: RegExp,
+    convertResponseToToken: (code: string, codeVerifierProxyPlaceholder: string) => Promise<OAuth2Result>,
+    options?: { timeout?: number, pollInterval?: number },
+  ): Promise<IOAuth2> {
+    const timeout = options?.timeout ?? (600 * 1000); // 10 minute timeout
```

Comment:
Very minor, but it would be nice if these values were put in a const. The same values are on L512.